processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperLogLogCollector.java

```diff
@@ -92,8 +92,23 @@ public static HyperLogLogCollector makeLatestCollector()
 
   public static HyperLogLogCollector makeCollector(ByteBuffer buffer)
   {
+    return makeCollector(buffer, false);
+  }
+
+  /**
+   * Create a wrapper object around an HLL sketch contained within a buffer.
+   *
+   * @param buffer      buffer containing HLL sketch
+   * @param bufferOwned if true, we are permitted to change the position and limit of the buffer
```

Comment:
Mentioned in private conversation:

The only reservation I would have is that there could be extensions which make the assumption of immutability. But as @gianm mentioned, having extensions call APIs that are not in druid-api is risky business. So I'm good with the proposed mutability change